101-jenkins-cluster-2-linux-1-win/azuredeploy.json

```diff
@@ -0,0 +1,596 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+                  "metadata": {
+                "description": "Location for cluster"
+            }
+        },
+
+        "master-vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A2_v2",
+            
+            "metadata": {
+                "description": "Master VM Size"
+            }
+        },
+        "linux-worker-vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A2_v2",
+            
+            "metadata": {
+                "description": "Linux worker VM Size"
+            }
+        },
+          "win-worker-vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_D2s_v3",
+            
+            "metadata": {
+                "description": "Windows Worker VM Size"
+            }
+        },
+        "master-username": {
+            "type": "string",
+            "defaultValue": "mahendra",
+            "metadata": {
+                "description": "Admin Username for Jenkins Master"
+            }
+        },
+        "node-username": {
+            "type": "string",
+            "defaultValue": "mahendra",
+            "metadata": {
+                "description": "Admin username for Slave Instances"
+            }
+        },
+        "master-password": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin Password for Jenkins Master VM"
+            }
+        },
+        "node-password": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for slave instances"
+            }
+        },
+        "jenkins-dns": {
+            "type": "string",
+            "defaultValue": "jenkins-cluster-mx",
```

Comment:
this must be globally unique, e.g.

[concat('jenkins-', uniqueString(resourceGroup().id))]